TaskHouseApi/Controllers/UsersController.cs

```diff
@@ -0,0 +1,124 @@
+        }
+
+        // GET: api/users 
+        // GET: api/users/?username=[username] 
+        [HttpGet]
+        public async Task<IEnumerable<User>> GetCustomers(string username)
+        {
+            if (string.IsNullOrWhiteSpace(username))
+            {
+                return await repo.RetrieveAllAsync();
+            }
+            else
+            {
+                return (await repo.RetrieveAllAsync())
+                .Where(user => user.Username == username);
+            }
+        }
+
+        // GET: api/users/[id] 
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetUser(int Id)
+        {
+            User u = await repo.RetrieveAsync(Id);
+            if (u == null)
+            {
+                return NotFound(); // 404 Resource not found 
+            }
+            return new ObjectResult(u); // 200 OK 
+        }
+
+        // POST: api/users
+        [AllowAnonymous]
+        [HttpPost]
+        public async Task<ActionResult<string>> CreateUser([FromBody]User user)
+        {
+            if (user == null)
+            {
+                return BadRequest(new { error = "CreateUser: user is null" }); // 400 Bad request 
+            }
+
+            User existingUser = (await repo.RetrieveAllAsync()).SingleOrDefault(u => u.Username == user.Username);
+
+            if (existingUser != null) {
+                return BadRequest(new { error = "Username in use" });
+            }
+
+            var hashResult = SecurityHandler.GenerateNewPassword(user);
+
+            user.Salt = hashResult.saltText;
+            user.Password = hashResult.saltechashedPassword;
+
+            User added = await repo.CreateAsync(user);
+            
+            return Ok();
+
+            //return CreatedAtRoute("GetUser", // use named route
+            //new { Id = added.Id }, user); // 201 Created
+        }
+
+        // PUT: api/users/[id] 
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Update(int Id, [FromBody] User u)
+        {
+            if (u == null || u.Id != Id)
+            {
+                return BadRequest(); // 400 Bad request 
+            }
+
+            var existing = await repo.RetrieveAsync(Id);
+            if (existing == null)
+            {
+                return NotFound(); // 404 Resource not found 
+            }
+
+            await repo.UpdateAsync(Id, u);
+            return new NoContentResult(); // 204 No content 
+        }
+
+        // DELETE: api/user/[id] 
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> Delete(int Id)
+        {
+            var existing = await repo.RetrieveAsync(Id);
+            if (existing == null)
+            {
+                return NotFound(); // 404 Resource not found 
+            }
+
+            bool deleted = await repo.DeleteAsync(Id);
+
+            if (deleted)
+            {
+                return new NoContentResult(); // 204 No content 
+            }
+            else
```

Comment:
else not needed, would suggest to invert the if it makes it easier to reason about and is good practice to have the success case in the bottom and all exceptions above.